codex/validation.nim

```diff
@@ -34,6 +34,12 @@ proc new*(
 proc slots*(validation: Validation): seq[SlotId] =
   validation.slots.toSeq
 
+proc iterateSlots(validation: Validation, action: proc(s: SlotId): Future[void] {.async.}) {.async.} =
```

Comment:
This pattern doesn't make sense and it will lead to even harder to track concurrency issue due to the introduction of async calls. 

The safer pattern is the most straightforward one, where you copy the `iterable` at the place of ussage.

Reply:
This will patch it: https://github.com/codex-storage/nim-codex/pull/809